ONSdigital/eq-author-app
# EAR 1960 Select and import pre-pop schema

### What is the context of this PR?

> Authors to be able to select and import pre-pop data
>

### How to review

> Update API gateway
- Cloud function `schemaVersionGet` has been added to project `author sds mock gateway`.  Check the following url returns data [api gateway](https://sds-api-mock-gateway-74jbrn8r.nw.gateway.dev/schemaVersionGet?id=121-222-789)

> Front end and Schema
- Check that the following is not present in the export schema prior to hitting a link button.  This is an example, the id will change depending on the row selected from the versions table which comes from mock versions data view here [Versions](https://sds-api-mock-gateway-74jbrn8r.nw.gateway.dev/schemaVersionsGet?survey_id=${121})

`"prepopSchema": {
"id": "122-111-789",
"schema": {
"id": "122-111-789",
"version": "1",
"dateCreated": "2023-01-12T13:37:27+00:00",
"turnover": {
"type": "number",
"example": "1000"
},
"employeeCount": {
"type": "number",
"example": "50"
}
}
}`

- Navigate to Data/ONS Dataset page
- Select a survey id from drop down
- Select Link from one of the table rows
- You should see a table showing linked data below
- Recheck exported data and check that the data in the first step has been added
- Select a different survey and select link from another row
- Linked data should update and show in the table and exported data
- Navigate away from page and back and the Linked data should be shown



### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
